go-app-sms_pmtct.js

```diff
@@ -0,0 +1,122 @@
+                return self.states.create("state_dial_not_sms");
+            } else {
+                // get the first word, remove non-alphanumerics, capitalise
+                switch (utils.get_clean_first_word(self.im.msg.content)) {
+                    case "STOP":
+                        return self.states.create("state_check_identity");
+                    default:
+                        return self.states.create("state_default");
+                }
+            }
+        });
+
+        self.states.add('state_check_identity', function(name) {
+            return is
+                .list_by_address({'msisdn': self.im.user.answers.msisdn})
+                .then(function(identities) {
+                    if (identities.results.length > 0) {
+                        self.im.user.set_answer('identity_id', identities.results[0].id);
+                        return self.states.create('state_opt_out_enter');
+                    } else {
+                        return self.states.create('state_end_unrecognised');
+                    }
+                });
+        });
+
+        self.states.add('state_dial_not_sms', function(name) {
+            return new EndState(name, {
+                text: $("Please use your handset's keypad to dial the number that you received, " +
+                        "rather than sending it to us in an sms."),
+                next: 'state_start',
+            });
+        });
+
+        self.states.add('state_opt_out_enter', function(name) {
+            var optout_info = {
+                optout_type: "stop",  // default to "stop"
+                identity: self.im.user.answers.identity_id,
+                reason: "unknown",  // default to "unknown"
+                address_type: "msisdn",  // default to 'msisdn'
+                address: self.im.user.answers.msisdn,
+                request_source: "sms_pmtct",
+                requestor_source_id: self.im.config.testing_message_id || self.im.msg.message_id,
+            };
+            return is
+                .optout(optout_info)
+                .then(function() {
+                    return self.states.create('state_opt_out');
+                });
+        });
+
+        self.states.add('state_opt_out', function(name) {
+            return new EndState(name, {
+                text: $('Thank you. You will no longer receive messages from us. ' +
+                        'If you have any medical concerns please visit your nearest clinic'),
+                next: 'state_start'
+            });
+        });
+
+        self.states.add('state_end_unrecognised', function(name) {
+            return new EndState(name, {
+                text: $("We do not recognise your number and can therefore not opt you out."),
+                next: 'state_start'
+            });
+        });
+
+        self.states.add('state_default', function(name) {
+            return new EndState(name, {
+                text: "default",
```

Comment:
Please replace `default` with `We do not recognise the message you sent us. Reply STOP to unsubscribe or dial {{channel}} for more options.` and replace channel with the the pmtct ussd line number (set it in the config).